Jenkinsfile

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env groovy
+
+pipeline {
+  agent {
+    label 'docker && linux'
+  }
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '5'))
+    timestamps()
+  }
+
+  stages {
+    stage('Build') {
+      environment {
+        JAVA_HOME = '/opt/jdk-17/'
+      }
+      steps {
+        sh './mvnw -V verify checkstyle:check spotbugs:check -Dmaven.test.failure.ignore -Dcheckstyle.failOnViolation=false -Dspotbugs.failOnError=false'
```

Comment:
this is correct. 
We don't want to stop the build process even when there are checkstyle violations or others. Same for the test. So that we don't have to commit the test fix, push, wait for the build, then see checkstyle issues, commit fix, push, wait, etc. etc.

Reply:
your understanding is perfect. 
The CI intends to validate any new addition on the source code. Yes, compiling is good, but f we have tests (unit tests or integration tests) then we can ensure that anything we add is not breaking the existing features set.

Reply:
Also, thanks to @lemeurherve and https://github.com/jenkins-infra/helpdesk/issues/3021#issuecomment-1170351247, know we cannot merge a pull request which is not validated by the CI.